src/main/java/life/qbic/datamodel/identifiers/SampleCodeFunctions.java

```diff
@@ -18,7 +18,13 @@
  */
 public class SampleCodeFunctions {
 
+  /** 
+   * @deprecated this is deprecated since 2.4.0 please use QBIC_SAMPLE_CODE_SCHEMA instead
+   */
+  @Deprecated
   public static final String QBIC_SAMPLE_ID_SCHEMA = "Q[A-X0-9]{4}[0-9]{3}[A-X0-9]{2}";
+  public static final String QBIC_SAMPLE_CODE_SCHEMA = "Q[A-X0-9]{4}[0-9]{3}[A-X0-9]{2}";
```

Comment:
Since `ENTITY_SAMPLES` are also `SAMPLES` I think we should choose a different name here. 
Or change `QBIC_ENTITY_SAMPLE_CODE_SCHEMA` to `QBOC_ENTITY_CODE_SCHEMA` in case we do not declare entity as samples.